.github/workflow_scripts/env_setup.sh

```diff
@@ -1,5 +1,5 @@
 function setup_build_env {
-    python -m pip install --upgrade pip
+    python -m pip install pip==21.3.1 # Capping pip to prevent tensorrt install failure
```

Comment:
Could you please provide some context regarding the need for version pinning here? An 'TODO' comment with additional details would be helpful.

Reply:
the proper fix would be to add a `PIP_EXTRA_INDEX_URL='https://pypi.nvidia.com'` environment variable to your CI environment (see the description [here](https://pypi.org/project/tensorrt/8.6.1.post1/))

for more details, see https://github.com/NVIDIA/TensorRT/pull/3080

Reply:
ah, at closer inspection of the failure, it does not seem to be an issue with tensorrt installation (the pip-inside-pip pattern of the tensorrt setup.py seems to be doing its job).

The flaky error was rather in some other custom wheel download code, namely from `nvidia_stub`:
```pytb
Traceback (most recent call last):
  File "/tmp/pip-build-env-yu1rd47j/overlay/lib/python3.10/site-packages/nvidia_stub/wheel.py", line 146, in download_wheel
    return download_manual(wheel_directory, distribution, version)
  File "/tmp/pip-build-env-yu1rd47j/overlay/lib/python3.10/site-packages/nvidia_stub/wheel.py", line 113, in download_manual
    raise RuntimeError(f"Didn't find wheel for {distribution} {version}")
RuntimeError: Didn't find wheel for nvidia-cudnn-cu12 9.0.0.312
```

That may have been a temporary host resolution error or so. In any case it looks like they're attempting to bypass the standardized install mechanisms provided by PyPA, probably for similar reasons as described in the PR discussion I linked above.

Reply:
Yup, it seemed to be temporary host resolution error.
Will look to add the proper fix if we see this failure again!
Thanks for the solution!